components/builder-db/src/models/projects.rs

```diff
@@ -95,6 +96,13 @@ impl Project {
         origin_projects::table.filter(origin_projects::origin.eq(origin))
                               .get_results(conn)
     }
+
+    pub fn count_origin_projects(origin: &str, conn: &PgConnection) -> QueryResult<i64> {
+        Counter::DBCall.increment();
+        origin_projects::table.select(count(origin_projects::id))
```

Comment:
Naive question: do we call `first` here because `select(count()).filter()` returns a single element Vec (or Vec-like) response?  My initial read made me think it might only ever count a single thing, but doesn't make a lot of sense.

Reply:
Great question @smacfarlane, I had to grok this too initially.
A couple of thoughts:

- According to the [docs](https://docs.diesel.rs/diesel/query_dsl/trait.RunQueryDsl.html#method.first) the behavior of `first`:
> Attempts to load a single record.This method is equivalent to .limit(1).get_result()

🤔 I suppose just `.get_result()` would also work but is not as explicit.

- I modeled the query after the existing [pub fn count_origin_packages](https://github.com/habitat-sh/builder/blob/46e8738ee38fda0e20f2ac33648e28a359c7db99/components/builder-db/src/models/package.rs#L632), my addition is basically a mirror of this one.